src/commands/yubihsm/keys/import.rs

```diff
@@ -0,0 +1,125 @@
+
+use serde_json::Value;
+
+/// The `yubihsm keys import` subcommand
+#[derive(Debug, Default, Options)]
+pub struct ImportCommand {
+    /// Path to the validator configuration file
+    #[options(short = "p", long = "path")]
+    pub path: Option<String>,
+
+    /// Type of key to generate (default 'ed25519')
+    #[options(short = "t")]
+    pub key_type: Option<String>,
+
+    /// Label for imported key
+    #[options(short = "l", long = "label")]
+    pub label: Option<String>,
+
+    /// Key ID for imported key
+    #[options(free)]
+    key_id: Option<u16>,
+}
+
+impl Callable for ImportCommand {
+    fn call(&self) {
+        if self.path.is_none() {
+            status_err!("must provide a valid path to priv_validator.json");
+            process::exit(1);
+        }
+
+        if self.key_id.is_none() {
+            status_err!("must provide a unique key_id");
+            process::exit(1);
+        }
+
+        match &self.key_type {
+            Some(ref key_type) => if key_type != DEFAULT_KEY_TYPE {
+                status_err!(
+                    "only supported key type is: ed25519 (given: \"{}\")",
+                    key_type
+                );
+                process::exit(1);
+            },
+            None => (),
+        }
+
+        if let Some(path) = &self.path {
+            let mut f = File::open(path).unwrap_or_else(|e| {
+                status_err!("couldn't open validator config file {}: {}", path, e);
+                process::exit(1);
+            });
+
+            let mut contents = Vec::new();
+            f.read_to_end(&mut contents).unwrap_or_else(|e| {
+                status_err!("couldn't read validator config file {}: {}", path, e);
+                process::exit(1);
+            });
+            let v: Value = serde_json::from_slice(&contents).unwrap();
+            let s = v["priv_key"]["value"].as_str().unwrap_or_else(|| {
+                status_err!("couldn't read validator private key from config: {}", path);
+                process::exit(1);
+            });
+
+            let key = base64::decode(s).unwrap_or_else(|e| {
+                status_err!("couldn't decode validator private key from config: {}", e);
+                process::exit(1);
+            });
+            Seed::from_keypair(&key).unwrap_or_else(|e| {
+                status_err!("invalid key in validator config: {}", e);
+                process::exit(1);
+            });
+            let label =
+                yubihsm::ObjectLabel::from(self.label.as_ref().map(|l| l.as_ref()).unwrap_or(""));
+
+            let mut hsm = yubihsm::get_hsm_client();
+
+            if let Err(e) = hsm.put_asymmetric_key(
+                self.key_id.unwrap(),
+                label,
+                DEFAULT_DOMAINS,
+                DEFAULT_CAPABILITIES,
+                yubihsm::AsymmetricAlg::Ed25519,
+                key,
+            ) {
+                status_err!("couldn't generate key #{}: {}", self.key_id.unwrap(), e);
```

Comment:
Should probably be `"couldn't import key"` or thereabouts

Reply:
Whoops. that parts was actually copy&pasted from the generate command (as they almost do the same thing).